File: scripts/compare_event_tables.js
#!/usr/bin/env node
const fs=require('fs'), vm=require('vm'), path=require('path');
const root=path.resolve(__dirname,'..');
const code=fs.readFileSync(path.join(root,'extension/core.js'),'utf8');
const sandbox={window:{},console,URL,Intl,Date,setTimeout,clearTimeout,btoa:(s)=>Buffer.from(s,'binary').toString('base64')};
vm.createContext(sandbox); vm.runInContext(code,sandbox);
const Core=sandbox.window.LDT_Core;
const raw=fs.readFileSync(path.join(root,'data/events.tsv'),'utf8');
const ev=Core.tsvToEvents(raw);
const out=Core.eventsToTSV(ev);
const ev2=Core.tsvToEvents(out);
if(ev.length!==ev2.length) throw new Error(`row mismatch ${ev.length} ${ev2.length}`);
for(let i=0;i<ev.length;i++){
  for(const k of ['title','lane','sub','href','timeZone','timeZoneLabel']){
    if(String(ev[i][k]||'')!==String(ev2[i][k]||'')) throw new Error(`field mismatch row ${i} ${k}`);
  }
}
console.log(`Node TSV roundtrip OK: ${ev.length} events`);

File: scripts/compare_event_tables.py
#!/usr/bin/env python3
import csv, hashlib, pathlib, sys
root=pathlib.Path(__file__).resolve().parents[1]
path=root/'data/events.tsv'
raw=path.read_text(encoding='utf-8').replace('\r\n','\n').strip()+"\n"
rows=list(csv.reader(raw.splitlines(), delimiter='\t'))
assert len(rows)>1, 'events.tsv has no rows'
cols=rows[0]
required=['title','shortTitle','category','lane','sub','start','endKnown','endInferred','href','timeZone','timeZoneLabel','isFixedTimeZone','status','source']
assert cols==required, f'header mismatch: {cols}'
for i,row in enumerate(rows[1:], start=2):
    assert len(row)==len(cols), f'line {i}: expected {len(cols)} columns, got {len(row)}'
    assert row[0] and row[5], f'line {i}: title/start required'
print('Python TSV structural comparison OK:', len(rows)-1, 'events,', len(cols), 'columns')
print('SHA256', hashlib.sha256(raw.encode('utf-8')).hexdigest())
